builder/hyperv/common/step_set_first_boot_device.go

```diff
@@ -0,0 +1,140 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepSetFirstBootDevice struct {
+	Generation      uint
+	FirstBootDevice string
+}
+
+func ParseBootDeviceIdentifier(deviceIdentifier string, generation uint) (string, uint, uint, error) {
+
+	captureExpression := "^(FLOPPY|IDE|NET)|(CD|DVD)$"
```

Comment:
It feels like the use of regex and checking against capture groups here really obfuscates the code. I'd rather we be a bit more verbose/less clever, maybe just with testing using the `strings` module so it's clearer to a naive user what is being done. 

I'd also really like to see a unit test that exercises this code path with valid and invalid inputs.

Reply:
The majority of the regular expressions have been replaced with a lookup table, using a map literal. Hopefully this will make the code less obtuse and increase its simplicity. 

Regular expressions are still used for `IDE:x:y` and `SCSI:x:y` because it is my opinion that using split and validating the various pieces would just be using the wrong tool for the job.